### PR TITLE
[WIP] feat: embed std/node polyfill into the binary

### PR DESCRIPTION
**Pull request by bartlomieju**
Experiment.

This commits creates a single bundle file from "std/node/module_all.ts" and
embeds it in "ext/node/03_std_node.js". This allows us to remove a lot of complexity
in CLI related to setting up Node compat layer. Thanks to being snapshotted we can
think about enabling this layer on each startup, which could help solve issues like https://github.com/denoland/deno/issues/16088.

Currently the produced bundle weighs at 1.8Mb. Compressed CLI snapshot weighs at 1093982 bytes, while on `main` it's 585684 bytes.

This should also make is significantly easier to support "node:<module_name>" imports in all code.

### CODE REVIEW

**Comment by ry** [comment]
Is it possible to use a git submodule?

**Comment by bartlomieju** [comment]
> Is it possible to use a git submodule? Or are you wanting to move the source of truth here?

Neither approach makes sense IMO - we will create a script that bundles `std/node` into `03_std_node.js`, because just pure bundling is not enough, we need to modify bundled file to "export" some things using "window.__bootstrap" namespace.

File: cli/module_loader.rs [hunk 1]
```diff
@@ -70,6 +70,24 @@ impl CliModuleLoader {
...
+        code: deno_runtime::deno_node::OS_ES_SHIM.to_string(),
+        found_url: specifier.to_owned(),
+        media_type: MediaType::JavaScript,
+      });
+    } else if specifier.as_str() == "node:tty" {
+      return Ok(ModuleCodeSource {
+        code: deno_runtime::deno_node::TTY_ES_SHIM.to_string(),
+        found_url: specifier.to_owned(),
+        media_type: MediaType::JavaScript,
+      });
```
Comment (cayter) [inline cli/module_loader.rs:90]: Do we also need to add `node:events` clause? I bumped into the error below when trying to print out `vite.middlewares` using Deno 1.29.1.

Screenshot 2023-01-08 at 5 58 55 PM

Comment (bartlomieju) [reply cli/module_loader.rs:90]: The issue you are running in will be resolved by https://github.com/denoland/deno/pull/17264

Comment (cayter) [reply cli/module_loader.rs:90]: Oh ok thanks.

**Comment by bartlomieju** [comment]
Closing in favor of https://github.com/denoland/deno/pull/17724

**Pull request closed without merging**